service.go

```diff
@@ -567,33 +621,91 @@ func (s *service) handleInvite(
 		go func() {
 			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 			defer cancel()
-			err := s.pullHistory(ctx, id, lg.ID)
-			if err != nil {
+			if err := s.pullHistory(ctx, id, lg.ID); err != nil {
 				log.Errorf("error pulling history: %s", err)
 			}
 		}()
 	}
 
 	// Create an own log if this is a new thread
-	var ownLog *thread.LogInfo
-	if invite.Readable() && newThread {
-		lg, err := util.CreateLog(s.threads.host.ID())
+	if logs.Readable() && newThread {
+		var lg thread.LogInfo
+		lg, err = util.CreateLog(s.threads.host.ID())
 		if err != nil {
-			return nil, err
+			return
 		}
 		err = s.threads.AddLog(id, lg)
 		if err != nil {
-			return nil, err
+			return
 		}
 		ownLog = &lg
 	}
 
+	// If not readable, return a new address for the sender's log.
+	// This peer becomes a follower.
+	if !logs.Readable() {
```

Comment:
This is a heavy handed assumption. We can revisit if a more explicit flow is needed.